test/budgie_web/live/budget_show_live_test.exs

```diff
@@ -107,13 +122,151 @@ defmodule BudgieWeb.BudgetShowLiveTest do
+               ~s|#transaction-modal input[name='transaction[description]'][value='#{transaction.description}']|
+             )
+    end
+
+    test "redirects on invalid transaction id", %{conn: conn, user: user, budget: budget} do
+      conn = log_in_user(conn, user)
+
+      {:ok, conn} =
+        live(conn, ~p"/budgets/#{budget}/transactions/invalid/edit")
+        |> follow_redirect(conn, ~p"/budgets/#{budget}")
+
+      assert %{"error" => "Transaction not found"} = conn.assigns.flash
+    end
+
+    test "validation errors are presented when form is changed with invalid input", %{
+      conn: conn,
+      user: user,
+      budget: budget,
+      transaction: transaction
+    } do
+      conn = log_in_user(conn, user)
+      {:ok, lv, _html} = live(conn, ~p"/budgets/#{budget}/transactions/#{transaction}/edit")
+
+      params = params_for(:budget_transaction, amount: Decimal.new("-42"))
+
+      form =
+        form(lv, "#transaction-modal form", %{
+          "transaction" => params
+        })
+
+      html = render_change(form)
+
+      assert html =~ "must be greater than 0"
+    end
+
+    test "validation errors are presented when form is submitted with invalid input", %{
+      conn: conn,
+      user: user,
+      budget: budget,
+      transaction: transaction
+    } do
+      conn = log_in_user(conn, user)
+      {:ok, lv, _html} = live(conn, ~p"/budgets/#{budget}/transactions/#{transaction}/edit")
+
+      params = params_for(:budget_transaction, amount: Decimal.new("-42"))
+
+      form =
+        form(lv, "#transaction-modal form", %{
           "transaction" => params
         })
 
       html = render_submit(form)
 
       assert html =~ "must be greater than 0"
     end
+
+    test "updates transaction", %{
+      conn: conn,
+      user: user,
+      budget: budget,
+      transaction: transaction
+    } do
+      conn = log_in_user(conn, user)
+      {:ok, lv, _html} = live(conn, ~p"/budgets/#{budget}/transactions/#{transaction}/edit")
+
+      new_amount = (:rand.uniform(5000) / 100) |> Decimal.from_float()
+
+      params = params_for(:budget_transaction, amount: new_amount)
+
+      form =
+        form(lv, "#transaction-modal form", %{
+          "transaction" => params
+        })
+
+      {:ok, _lv, html} =
+        render_submit(form)
+        |> follow_redirect(conn)
+
+      assert html =~ "Transaction updated"
+
+      transaction = Repo.get(BudgetTransaction, transaction.id)
+      assert transaction.amount == new_amount
+    end
+  end
+
+  describe "Delete transaction" do
+    setup %{budget: budget} do
+      transaction = insert(:budget_transaction, budget: budget)
+
+      %{transaction: transaction}
+    end
+
+    test "deletes transaction", %{
+      conn: conn,
+      user: user,
+      budget: budget,
+      transaction: transaction
+    } do
+      conn = log_in_user(conn, user)
+      {:ok, lv, html} = live(conn, ~p"/budgets/#{budget}")
+
+      assert html =~ transaction.description
```

Comment:
The assertion of removal below only makes sense if we can guarantee that the description was present in the pre-deletion render.